pythonforandroid/bootstraps/sdl2/build/src/main/java/org/kivy/android/PythonActivity.java

```diff
@@ -364,118 +385,219 @@ public static void stop_service() {
+            } else if (PythonActivity.mImageView.getParent() == null) {
+                mLayout.addView(mImageView);
+            }
+        } catch (IllegalStateException e) {
+            // The loading screen can be attempted to be applied twice if app
+            // is tabbed in/out, quickly.
+            // (Gives error "The specified child already has a parent.
+            // You must call removeView() on the child's parent first.")
+        }
     }
     
     @Override
     protected void onPause() {
-    	// fooabc
-        if ( this.mWakeLock != null &&  mWakeLock.isHeld()){
-        	this.mWakeLock.release();
+        if (this.mWakeLock != null && mWakeLock.isHeld()) {
+            this.mWakeLock.release();
         }
 
         Log.v(TAG, "onPause()");
-        super.onPause();
+        try {
+            super.onPause();
+        } catch (UnsatisfiedLinkError e) {
+            // Catch pause while still in loading screen failing to
+            // call native function (since it's not yet loaded)
+        }
     }
 
     @Override
     protected void onResume() {
-    	if ( this.mWakeLock != null){
-    		this.mWakeLock.acquire(); 
-    	}
-	    Log.v(TAG, "onResume()");
-	    super.onResume();
+        if (this.mWakeLock != null) {
+            this.mWakeLock.acquire();
+        }
+        Log.v(TAG, "onResume()");
+        try {
+            super.onResume();
+        } catch (UnsatisfiedLinkError e) {
+            // Catch resume while still in loading screen failing to
+            // call native function (since it's not yet loaded)
+        }
+        considerLoadingScreenRemoval();
     }
 
-    
+    @Override
+    public void onWindowFocusChanged(boolean hasFocus) {
+        try {
+            super.onWindowFocusChanged(hasFocus);
+        } catch (UnsatisfiedLinkError e) {
+            // Catch window focus while still in loading screen failing to
+            // call native function (since it's not yet loaded)
+        }
+        considerLoadingScreenRemoval();
+    }    
 
+    /**
+     * Used by android.permissions p4a module to check a permission
+     **/
+    public boolean checkCurrentPermission(String permission) {
+        if (android.os.Build.VERSION.SDK_INT < 23)
+            return true;
+
+        try {
+            java.lang.reflect.Method methodCheckPermission =
+                Activity.class.getMethod("checkSelfPermission", java.lang.String.class);  
+            Object resultObj = methodCheckPermission.invoke(this, permission);
+            int result = Integer.parseInt(resultObj.toString());   
+            if (result == PackageManager.PERMISSION_GRANTED) 
+                return true;
+        } catch (IllegalAccessException | NoSuchMethodException |
+                 InvocationTargetException e) {
+        }
+        return false;
+    }
+
+    /**
+     * Used by android.permissions p4a module to request a permission
+     **/
+    public void requestNewPermission(String permission) {
```

Comment:
I've started playing with this recently and it seems to work, thanks for this @JonasT 
Any reason why we didn't stick to official API signature? I mean why isn't this called `requestPermissions` and why doesn't it take a list of permission string?

Reply:
@AndreMiras oh right that would have been smarter I guess :grimacing:

Reply:
Thanks @JonasT :pray: 
I just wanted your feedback on it. I've created a ticket to address it
https://github.com/kivy/python-for-android/issues/1704